app/src/main/java/de/westnordost/streetcomplete/quests/NameAndLocationLabel.kt

```diff
@@ -0,0 +1,116 @@
+package de.westnordost.streetcomplete.quests
+
+import android.content.res.Resources
+import android.text.Html
+import android.text.Spanned
+import androidx.core.os.ConfigurationCompat
+import androidx.core.text.parseAsHtml
+import de.westnordost.osmfeatures.FeatureDictionary
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.ktx.toList
+import java.util.Locale
+
+fun Resources.getNameAndLocationLabelString(tags: Map<String, String>, featureDictionary: FeatureDictionary): Spanned? {
+    val localeList = ConfigurationCompat.getLocales(configuration).toList()
+    val featureName = getFeatureName(tags, featureDictionary, localeList)
+        ?.withNonBreakingSpaces()?.inItalics()
+    val nameLabel = getNameLabel(tags)?.withNonBreakingSpaces()?.inBold()
+    val levelLabel = getLevelLabel(tags)
+    val houseNumberLabel = getHouseNumberLabel(tags)
+
+    val result = mutableListOf<String>()
+    if (levelLabel != null) {
+        result += levelLabel
+    }
+    // only show housenumber if there is neither name nor level information
+    if (levelLabel == null && nameLabel == null && houseNumberLabel != null) {
+        result += houseNumberLabel
+    }
+    // if both name and feature name are available, put the feature name in parentheses
+    if (nameLabel != null) {
+        result += nameLabel
+        if (featureName != null) result += "($featureName)"
+    } else {
+        if (featureName != null) result += "$featureName"
+    }
+
+    return if (result.isEmpty()) null else result.joinToString(" ").parseAsHtml()
```

Comment:
With the label below the title, this needs a bit more logic to show/hide the colon:

![grafik](https://user-images.githubusercontent.com/202916/158059819-f527f0c2-43db-4f01-ad7d-fd1cc4ee5efd.png)

e.g. for https://www.openstreetmap.org/way/132149975

Reply:
ok